IBX-281: Enable binary file fields indexing (#1386)
ezsystems/developer-documentation: docs/guide/file_management.md

For clustering use we provide a custom metadata handler that stores metadata about your assets in the database. This is done as it is faster than accessing the remote NFS or S3 instance in order to read metadata. For further reading on setting this up, see [Clustering](clustering.md).

## Enable BinaryFile Field indexing

The indexing of all binary file Fields is disabled by default.
If you want to enable indexing, you must have installed Oracle Java/Open JDK 8 or higher and Apache Tika 1.20.
Next, in the `config/packages` folder create a `binary_files.yaml` file with the following configuration:

``` yaml
ibexa_platform_commerce_field_types:
    binary_file_indexing:
        enabled: true
```

> To check what types are indexed by default, see under the `siso_search.<site_access>.index_content` parameter in `src/Siso/Bundle/SearchBundle/Resources/config/search.yml`. This parameter can be overriden, so you use it to index only specific types per SiteAccess or to extend the indexing to other file types.

Reviewer: ```suggestion
To check what types are indexed by default, see under the `siso_search.default.index_content` parameter in `src/Siso/Bundle/SearchBundle/Resources/config/search.yml`. This parameter can be overriden, so you use it to index only specific types per SiteAccess or to extend the indexing to other file types.
```

but:

Maybe can just list those indexed by default,
```
        - application/pdf
        - application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
```
 and mention it can be overriden with `siso_search.default.index_content`

For clustering use we provide a custom metadata handler that stores metadata about your assets in the database. This is done as it is faster than accessing the remote NFS or S3 instance in order to read metadata. For further reading on setting this up, see [Clustering](clustering.md).

## Enable BinaryFile Field indexing

> The indexing of all binary file fields is disabled by default. If you want to enable indexing, you must have installed Oracle Java/Open JDK 8 or higher and Apache Tika 1.20.

Reviewer 2: ```suggestion
The indexing of all binary file Fields is disabled by default.
If you want to enable indexing, you must have installed Oracle Java/Open JDK 8 or higher and Apache Tika 1.20.
```

> Next, in the `config/packages/` create a `<name>.yaml` file with the following configuration:

Reviewer 2: ```suggestion
Next, in the `config/packages` folder create a `<name>.yaml` file with the following configuration:
```

> Next, in the `config/packages/` create a `<name>.yaml` file with the following configuration:

Reviewer 2: Unless we can suggest a name here? something like:
```suggestion
Next, in the `config/packages` folder create a `binary_files.yaml` file with the following configuration:
```

``` yaml
ibexa_platform_commerce_field_types:

>   binary_file_indexing:
>     enabled: true

Reviewer 2: ```suggestion
    binary_file_indexing:
        enabled: true
```
Just for consistency with other examples in doc.

```

> To check what file types are indexed, see under the `siso_search.<site_access>.index_content` parameter in `src/Siso/Bundle/SearchBundle/Resources/config/search.yml`. 

Reviewer: Those file types are indexed by default (in `siso_search.default.index_content`), that param can be overriden, if you want to extend indexing to other file types, or be used to index only specific types per SA. Like in, `en` SA you want to index only pdf files for some reason.

> 
> 
> 

Reviewer 2: ```suggestion

```
